src/main/fc/init.c

```diff
@@ -391,16 +392,16 @@ void init(void)
     initInverters(serialPinConfig());
 #endif
 
+    ioPreInitAll();
```

Comment:
@MJ666 Oh, btw, `targetBusInit` will be gone in generic target, too.

Reply:
I wouldn't be making assumptions about overloading of pins - ribbon cable connectors for peripherals are becoming more and more common, and there are cases of peripherals using a UART on the same connector that also carries SPI, so there will be some overloading happening.

Reply:
Yes. You are right. My NXT does that...

Reply:
Wait. Overloading in hardware possibilities do happen, but in software config, it doesn't (shouldn't) happen. What I mean by pin overloading is something that needs automatic device detection.

Something like this is happening everywhere. It is user's responsibility to avoid actual collision of overloaded function; not to enable UART functions if corresponding pins are used for motors or I2C and vice versa.

Reply:
Interrupt pins are input pins and can be initialized by the drive as it is today. IMHO there is no need fro preinitializing them? In my understanding (F1 and) F3 are not going to be generic targets? The targetBusInit is only used there. how this should be replaced?

Reply:
@jflyper: Ah, now I understand. Yes, sharing of a pin between functions at runtime should not happen (except for the well defined cases like UART sharing for MSP / something else).

Re need for preinit, I am not sure if there is a use case for this outside of chip select signals on bussed devices - in general a device needs to be hooked up in a way that makes it support the default pin signal where it is hooked up before its initialisation code is run, so there should not be need to do anything before initialisation. The exception there is CS, where we need to make sure all devices sharing a bus are kept quiet while there is activity on the shared bus during the one by one device initialisation.

Reply:
Renamed everything back to `spiPreinitXXX`, `spiPreinit` call in `init` is now inside `USE_SPI` and `spiPreinit` call from `targetBusInit` restored.